### ISSUE TITLE
New feature to allow deleting of all of a user's given thanks/likes

### ISSUE THREAD

**Issue by lairdshaw** [mybbgroup/MyBB_Thank-you-like-plugin#198]
This feature was requested by floriaN [here](https://community.mybb.com/thread-169382-post-1309715.html#pid1309715):

> Is there a query or feature to wipe/remove all likes from a User? Like setting his likes to 0.

And further explained [here](https://community.mybb.com/thread-169382-post-1309780.html#pid1309780):

> I have members abusing the like system with creating multi accounts and liking their own post to reach a new usergroup or to bypass limitation. Perm banning them is not the solution here, I'd rather just delete all of their likes as a punishment.

floriaN has offered to pay me to develop this feature at a mutually-agreed price - the feature would then be available as usual as part of the freely-licensed public version of the plugin. This scenario has been OKed by Eldenroot so long as the feature is first discussed publicly as a GitHub issue - and this is it.

So, here's my suggestion as to how to implement this feature:

Add a new moderation permission which can be toggled on/off for any given usergroup via a checkbox, "Can delete users' thanks/likes?", at the bottom of the "Moderator CP" tab when editing a usergroup.

When viewing a member's profile, members of a usergroup with this option toggled on will be shown an additional link in the "Moderator Options" pane: "Delete thanks/likes given by this user".

When viewing the Mod CP, members of a usergroup with this option toggled on will be shown an additional link in the left navigation panel under "Users": "Delete Thanks/Likes By" (with the thumbs-up image as its icon).

Clicking on either of these links will lead to a page with a username field (pre-filled when arriving via a member profile). Filling in a valid username (autocomplete will as usual be enabled) and clicking "Delete" will delete all thanks/likes given by the user.

And that's it. Let me know what you think!

**Comment by Eldenroot** [comment]
I agree, good approach! Feel free to work on it.

Is possible to send a PM to user automatically when his thanks/likes were deleted with reason?

**Comment by lairdshaw** [comment]
Yes, an automated PM would be possible - good idea, let's add it. Something else to add is logging the action to the moderator log.

**Comment by Eldenroot** [comment]
Yes, logging is useful.

You can start with coding and during that there might be more ideas what to add or improve.

**Comment by lairdshaw** [comment]
An overdue update: floriaN and I couldn't agree on a price. I no longer plan to implement this feature - it's open to anybody else to do.

**Comment by Eldenroot** [comment]
@SvePu - please check this branch, it is not correctly integrated (because we have to handle deletion thx/likes in a correct way which will not bug the recounting thx/likes)... anyway could be used somehow maybe? 

https://github.com/Eldenroot/thankyoulikeold/commit/b05d5ae4efcd22de83827403d4b0b962ff558e9c

**Comment by SvePu** [comment]
I'll take a look

**Comment by Eldenroot** [comment]
Thank you!

**Comment by SvePu** [comment]
@Eldenroot Why you don't use an adapted thankyoulike_delete_user() fuction??
https://github.com/mybbgroup/Thank-you-like-system/blob/fbe254f34b69ba954d512c2725a19d2af7dcb833/upload/inc/plugins/thankyoulike.php#L3294-L3360

**Comment by Eldenroot** [comment]
Do not ask me :D this was just a try... hidden in my private repo :) feel free to adjust it or whatever (or ignore it) :)
